rmf_traffic/test/unit/agv/test_Planner.cpp

```diff
@@ -458,6 +458,159 @@ SCENARIO("Test Options", "[options]")
     CHECK(*default_options.interrupt_flag());
   }
 
+  Planner::Options default_maxcostestimate(nullptr);
+  WHEN("Maximum cost estimate is default after default construction")
+  {
+    // Converting an optional to a bool returns false if it's empty, which is
+    // the default for constructing Options
+    CHECK(!default_maxcostestimate.maximum_cost_estimate());
+  }
+
+  double local_maximum_cost_estimate(3.14159);
+  Planner::Options set_maxcostestimate(
+    nullptr,
+    Planner::Options::DefaultMinHoldingTime,
+    interrupt_flag,
+    local_maximum_cost_estimate);
+  WHEN("Maximum cost estimate is set after construction")
+  {
+    CHECK(set_maxcostestimate.maximum_cost_estimate().value()
+      == Approx(local_maximum_cost_estimate).margin(1e-6));
```

Comment:
```suggestion
      == Approx(local_maximum_cost_estimate));
```

The default margin should be okay here since we're expecting an exact binary match between these values.